wgsl/index.bs

```diff
@@ -422,6 +422,95 @@ The result type depends on the number of letters provided. Assuming a `vec4<f32>
   </xmp>
 </div>
 
+# Textures # {#textures}
+
+## Texture Types ## {#texture-types}
+
+* type must be `f32`, `i32` or `u32`
+* The parameterized type for the images is the type after conversion from reading/sampling. E.g. you can have an image with texels with 8bit unorm components, but when you sample them you get a 32-bit float result (or vec-of-f32).
+
+### Sampled Texture ### {#texture-sampled}
+
+<pre class='def'>
+`texture_sampled_1d<type>`
+  %1 = OpTypeImage %type 1 0 0 0 1 Unknown
+
+`texture_sampled_2d<type>`
+  %1 = OpTypeImage %type 2 0 0 0 1 Unknown
+
+`texture_sampled_3d<type>`
+  %1 = OpTypeImage %type 3 0 0 0 1 Unknown
+
+`texture_sampled_2d_ms<type>`
+  %1 = OpTypeImage %type 2 0 0 1 1 Unknown
+</pre>
+
+### Read-only Storage Texture ### {#texture-ro}
+<pre class='def'>
+`texture_ro_1d<type>`
+  %1 = OpTypeImage %type 1 0 0 0 2 Unknown
+
+`texture_ro_2d<type>`
+  %1 = OpTypeImage %type 2 0 0 0 2 Unknown
+
+`texture_ro_3d<type>`
+  %1 = OpTypeImage %type 3 0 0 0 2 Unknown
+
+`texture_ro_2d_ms<type>`
+  %1 = OpTypeImage %type 2 0 0 1 2 Unknown
+</pre>
+
+### Write-only Storage Texture ### {#texture-wo}
+<pre class='def'>
+`texture_wo_1d<type>`
+  %1 = OpTypeImage %type 1 0 0 0 2 Unknown
+
+`texture_wo_2d<type>`
+  %1 = OpTypeImage %type 2 0 0 0 2 Unknown
+
+`texture_wo_3d<type>`
+  %1 = OpTypeImage %type 3 0 0 0 2 Unknown
+
+`texture_wo_2d_ms<type>`
+  %1 = OpTypeImage %type 2 0 0 1 2 Unknown
+</pre>
+
+### Depth Texture ### {#texture-depth}
+<pre class='def'>
+`texture_depth_2d<type>`
+  %1 = OpTypeImage %type 2 0 0 0 1 Unknown
+</pre>
+
+### Samplers ### {#texture-sampler}
+<pre class='def'>
+sampler
+  OpTypeSampler
+
+sampler_comparison
+  OpTypeSampler
+</pre>
+
+## Texture Functions ## {#texture-functions}
+
+<pre class='def'>
+vec4<type> texture_load(texture_ro, f32 | vec(2|3)<f32> coords, i32 level_of_detail)
```

Comment:
@dneto0 do we really need different methods here? Since the semantics of them is the same, we could have the same `texture_load()` function working (overloaded) with both `texture_sampled*` and `texture_ro`. Would this be a concern?

Reply:
Added todos.

Reply:
Overloading on the texture type works for me.